test/openid_connect/worker_test.exs

```diff
@@ -98,4 +96,10 @@ defmodule OpenIDConnect.WorkerTest do
     end)
     |> expect(:get, fn "https://www.googleapis.com/oauth2/v3/certs", _headers, _opts -> @google_certs end)
   end
+
+  defp start_worker(config) do
+    {:ok, pid} = start_supervised({OpenIDConnect.Worker, config})
+    Process.sleep(10) # allow :update_documents to run
```

Comment:
Doing sleep during a test isn't ideal. It would be better if the `pid` of the test could be added as an option and if present the worker would notify the pid when the document was updated. Then the test would `assert_received` for that message and timeout/fail if it doesn't happen.

Reply:
I agree. The test doesn't really need to be notified when update_documents completes (it's mocked), but the `:update_documents` should be in the worker's queue before the test continues.

Reply:
as it stands, I think the PR's architecture implementation is wrong. The update_documents function should be blocking. I'd like to see failing test cases and work back from there to a solution

Reply:
`update_documents` is still blocking. It was already a `handle_info` before this PR, I haven't changed that, and it schedules itself periodically, as before. I deferred it's invocation from `init/1` to allow the worker (and it's supervision tree) to remain alive in the case of a temporary network error (domain resolution failure, for example). 

Feel free to look at alternatives (#30).